File: src/003/p003.go
package main

import "fmt"

func solve() int {
	n := 600851475143
	limit := n / 2
	i := 3

	for i < limit {
		for n%i == 0 {
			n = n / i
			limit = n / 2
		}
		i += 2
	}
	return n
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/003/p003.lua
function solve()
  local n = 600851475143
  local limit = n / 2
  local i = 3

  while i < limit do
    while n % i == 0 do
      n = n / i
      limit = n / 2
    end
    i = i + 2
  end
  return math.floor(n)
end


local result = solve()
print(result)
